add ili9341-2

gemu2015 opened the pull request:
## Description:

support for ILI9341 with support for 2. Hardware SPI BUS needed by TTGO-T4
since Tasmota supports only 1 Hardware SPI BUS the second has to be defined by SSPI. On ESP32 the 2. Hardware SPI is used on ESP8266 software SPI
remark:
the TTGO-T4 is a very nice all in One ESP32 board which is easy to expand since a lot of pins are externally available and and also contains a grid circuit board for your own expansions
<img width="400" alt="Bildschirmfoto 2020-11-16 um 06 47 44" src="https://user-images.githubusercontent.com/11647075/99217139-a5ea9c00-27d7-11eb-9146-7dd3227ce8e3.png">


missing dimmer init added 

## Checklist:
  - [x] The pull request is done against the latest dev branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR.
  - [x] The code change is tested and works on Tasmota core ESP8266 V.2.7.4.7
  - [x] The code change is tested and works on Tasmota core ESP32 V.1.0.4.2
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

gemu2015 committed:
add ili9341-2

jeroenvermeulen commented:
@gemu2015 Can you tell me the pins or template? I have this device.

jeroenvermeulen commented:
I have found the [original demo software](https://github.com/Xinyuan-LilyGO/LilyGo_Txx) and got it working. They are using the [TFT_eSPI](https://github.com/Bodmer/TFT_eSPI) library.

Pins for TTGO T4 v1.3 are:
```
#define TFT_MISO 12  // aka SDO
#define TFT_MOSI 23  // aka SDI
#define TFT_SCLK 18  //
#define TFT_CS   27  // 
#define TFT_DC   32  // aka RS
#define TFT_RST   5  // OLED Reset
#define TFT_BL    4  // Backlight

#define SD_MISO     2
#define SD_MOSI     15
#define SD_SCLK     14
#define SD_CS       13

#define I2C_SDA     21
#define I2C_SCL     22

#define SPEAKER_PWD 19
#define SPEAKER_OUT 25
#define ADC_IN      35

#define BUTTON_1    37
#define BUTTON_2    38
#define BUTTON_3    39
#define BUTTON_4    0
```

gemu2015 commented:
this device does not have a speaker. since i like to have sound on my devices i added an i2s audio amp and a small speaker in the bottom and removed the battery which i don't need.

the sd card was only usable with scripts because scripts add extended file system support.
if you want to test SD card without scripts you may try my forks driver xdrv98 which adds a universal file system to tasmota.

#define USE_UFILESYS
#define USE_SDCARD
#define SDCARD_CS_PIN 13

but currently there is not much use from rules.  with i2s audio you could use the cmds play, say, and time 
from rules.
